src/arb/arb.jl

```diff
@@ -85,19 +85,86 @@ characteristic(::ArbField) = 0
 #
 ################################################################################
 
-function Float64(x::arb)
-   GC.@preserve x begin
-      t = ccall((:arb_mid_ptr, libarb), Ptr{arf_struct}, (Ref{arb}, ), x)
-      # 4 == round to nearest
-      d = ccall((:arf_get_d, libarb), Float64, (Ptr{arf_struct}, Int), t, 4)
-   end
-   return d
+@doc Markdown.doc"""
+    Float64(x::arb, round::RoundingMode=RoundNearest)
+
+Converts $x$ to a `Float64`, rounded in the direction specified by $round$.
+For `RoundNearest` the return value approximates the midpoint of $x$. For
+`RoundDown` or `RoundUp` the return value is a lower bound or upper bound for
+all values in $x$.
+"""
+function Float64(x::arb, round::RoundingMode=RoundNearest)
+  t = _arb_get_arf(x, round)
+  return _arf_get_d(t, round)
+end
+
+@doc Markdown.doc"""
+    BigFloat(x::arb, round::RoundingMode=RoundNearest)
+
+Converts $x$ to a `BigFloat` of the currently used precision, rounded in the
+direction specified by $round$. For `RoundNearest` the return value
+approximates the midpoint of $x$. For `RoundDown` or `RoundUp` the return
+value is a lower bound or upper bound for all values in $x$.
+"""
+function BigFloat(x::arb, round::RoundingMode=RoundNearest)
+  t = _arb_get_arf(x, round)
+  return _arf_get_mpfr(t, round)
+end
+
+function _arb_get_arf(x::arb, ::RoundingMode{:Nearest})
+  t = arf_struct()
+  GC.@preserve x begin
+    t1 = ccall((:arb_mid_ptr, libarb), Ptr{arf_struct},
+               (Ref{arb}, ),
+               x)
+    ccall((:arf_set, libarb), Nothing,
+          (Ref{arf_struct}, Ptr{arf_struct}),
+          t, t1)
+  end
+  return t
+end
+
+for (b, f) in ((RoundingMode{:Down}, :arb_get_lbound_arf),
+               (RoundingMode{:Up}, :arb_get_ubound_arf))
+  @eval begin
+    function _arb_get_arf(x::arb, ::$b)
+      t = arf_struct()
+      ccall(($(string(f)), libarb), Nothing,
+            (Ref{arf_struct}, Ref{arb}, Int),
+            t, x, parent(x).prec)
+      return t
+    end
+  end
+end
+
+for (b, i) in ((RoundingMode{:Down}, 2),
+               (RoundingMode{:Up}, 3),
+               (RoundingMode{:Nearest}, 4))
+  @eval begin
+    function _arf_get_d(t::arf_struct, ::$b)
+      d = ccall((:arf_get_d, libarb), Float64,
+                (Ref{arf_struct}, Int),
+                t, $i)
+      return d
+    end
+    function _arf_get_mpfr(t::arf_struct, ::$b)
+      d = BigFloat()
```

Comment:
Or rather here

Reply:
I see no problem. The precision is read from the global variable by default in mpfr.jl
```
    function BigFloat(; precision::Integer=DEFAULT_PRECISION[])
...
```

Reply:
Is the current version of `BigFloat(x::arb)` given the result with `DEFAULT_PRECISION[]` or not?

Reply:
The current version of FigFloat(::arb) uses DEFAULT_PRECISION[] as we just discussed. Here is the pudding for some proof:
```
julia> x = const_pi(ArbField(110))
[3.14159265358979323846264338327950 +/- 4.34e-33]

julia> BigFloat(x)
3.141592653589793238462643383279502797479068098137295573004504331874296718662976

julia> precision(ans)
256

julia> setprecision(100) do; BigFloat(x); end
3.1415926535897932384626433832793

julia> precision(ans)
100
```

Reply:
nice